lib/OperationHandler.cpp

```diff
@@ -433,31 +453,39 @@ ScanResponse OperationHandler::HandleScanRequestSerialized(const ScanRequest& sc
         scanRequest->ssid_len > 0 ? std::make_optional<const Ssid>(gsl::span{scanRequest->ssid, scanRequest->ssid_len}) : std::nullopt;
 
     // Start all scan requests
-    std::vector<std::future<std::vector<ScannedBss>>> futureScanResults;
+    using FutureScanResult = std::future<std::pair<std::vector<ScannedBss>, ScanStatus>>;
+    std::vector<std::pair<GUID, FutureScanResult>> futureScanResults;
     for (const auto& wlanIntf : m_wlanInterfaces)
     {
         try
         {
-            futureScanResults.push_back(wlanIntf->Scan(requestedSsid));
+            futureScanResults.emplace_back(wlanIntf->GetGuid(), wlanIntf->Scan(requestedSsid));
         }
         CATCH_LOG_MSG("WlanScan failed")
     }
 
     // Collect all scan results and merge them
     ScanResponseBuilder scanResponse;
     const auto timeout = std::chrono::steady_clock::now() + std::chrono::seconds(10);
-    for (auto& scanFuture : futureScanResults)
+    for (auto& [intfGuid, scanFuture]: futureScanResults)
     {
         try
         {
             std::vector<ScannedBss> scanResults;
+            auto status = ScanStatus::Completed;
             if (scanFuture.wait_until(timeout) != std::future_status::ready)
             {
                 LOG_WIN32_MSG(ERROR_TIMEOUT, "Scan timed out.");
```

Comment:
I might be mis-understanding this bit. `status` is set to `Completed` above, but then if the `wait_until` times out, the results are still added to `scanResponse` below. 

So, is the idea here that you'll wait some time for the scan to complete, but if it doesn't, you'll provide the results so far?

Reply:
I am actually cheating a bit here.
The initial design was to set `status = Completed` only if all interfaces returned `Completed`. If not completed here, the scan would be completed later through a notification.

In practice, in the guest, wpa_supplicant tends to not cancel the scan when it isn't completed. Since our goal is avoid waiting for a real host scan completion, I lie here and always pretend the scan is completed. If it isn't, the full results will still be sent through a notification. If a good enough result isn't present in the uncomplete results, the guest will schedule a new scan (and this time, we will wait for the actual results).

I will add a comment to clarify.

Reply:
Actually, ignore my previous comment, I was misremembering.
If waiting for the future times out, I don't have results from this scan and can't assume I'll ever get some. The result list will be empty, so I am not adding anything, and I am not adding the interface to `m_scanningInterfaces` since I didn't even got a "I am busy scanning" answer, but simply no answer at all.

Reply:
> The result list will be empty, so I am not adding anything

Ok, I think I understand. The code to process the results still runs (lines 492-495) but `scanResult.bssList` is empty since there was a timeout, so no results will be effectively added. I was a bit confused because only on line 483 does `scanResult` get set to those from the future (which makes sense), and it didn't click that the default-constructed `IWlanInterface::ScanResult scanResult` was empty.